visualizations/frontend/filters/v2/TDatePicker.vue

```diff
@@ -0,0 +1,282 @@
+                    break;
+                
+                case "last30days":
+                    startDate = new Date(startDate.setDate(startDate.getDate() - 30));
+                    endDate = new Date(endDate.setDate(endDate.getDate() - 1));
+                    break;
+                
+                case "last90days":
+                    startDate = new Date(startDate.setDate(startDate.getDate() - 90));
+                    endDate = new Date(endDate.setDate(endDate.getDate() - 1));
+                    break;
+                
+                case "mtd":
+                    startDate.setDate(1);
+                    break;
+                
+                case "ytd":
+                    startDate.setMonth(0);
+                    startDate.setDate(1);
+                    break;
+            }
+
+            this.date = [startDate, endDate];
+        },
+        handleSingleDates(preset) {
+            let date = new Date();
+            switch(preset) {
+                case "yesterday":
+                    date.setDate(date.getDate() - 1);
+                    break;
+
+                case "7DaysAgo":
+                    date.setDate(date.getDate() - 7);
+                    break;
+
+                case "30DaysAgo":
+                    date.setDate(date.getDate() - 30);
+                    break;
+                
+                case "90DaysAgo":
+                    date.setDate(date.getDate() - 90);
+                    break;
+                
+                case "yearAgo":
+                    date.setDate(date.getDate() - 365);
+                    break;
+                
+            }
+            this.date = date;
+        },
+        handleClear() {
+            this.date = [];
+            this.selectedPreset = {};
+            this.unsetFilterValue("start_date");
+            this.unsetFilterValue("end_date");
+            this.unsetFilterValue("date_preset");
+            this.unsetFilterValue("date");
+        },
+        formatDate(dateValue) {
+            const date = [];
+
+            const dateObject = new Date(dateValue);
+            let year = dateObject.getFullYear();
+
+            let month = dateObject.getMonth() + 1;
+            month = month < 10 ? ("0" + month) : month;
+
+            let day = dateObject.getDate();
+            day = day < 10 ? ("0" + day) : day;
+
+            let formatItems = this.dateFormat.match(/(.)\1*/g);  // Formats the date-format like this ['YYYY', '-', 'MM', '-', 'DD']
+            for (let item of formatItems) {
+                if (item === "YYYY") {
+                    date.push(year);
+                } else if (item === "MM") {
+                    date.push(month);
+                } else if (item === "DD") {
+                    date.push(day);
+                } else {
+                    // It's a seperator
+                    date.push(item);
+                }
+            }
+            return date.join("");
```

Comment:
Do you have access to `moment` here? It might be helpful to use it for formatting dates. 

```suggestion
    return moment().format('YYYY-MM-DD')
```

Reply:
Yes, I can access moment here but @egcastro had some opinion on moment I think, basically moment library is bloated and I think is no longer getting updates, Emi can confirm.

Reply:
I presume that Emi was going to suggest using `date-fns`. From what I have seen, `date-fns` has a very similiar API to moment. However It's not a problem to format the dates by hand if you want to avoid dependencies.

Reply:
Exactly, we can use moment.js on this PR but track it so we can get rid of it when we switch over to `date-fns`